### PR TITLE
Update component “tables”

### PR DESCRIPTION
**Pull request by itsmemin**
Automatically generated by Netlify CMS

### CODE REVIEW

File: src/_areas/design-system/components/tables.md [hunk 1]
```diff
@@ -11,75 +11,231 @@ statusOverride: null
 ---
 ## Usage
 
-Use tables to help users visualize, organize and understand their data to derive insights and take informed action.  
+Used to help users visualize, organize and understand their data to derive insights and take informed action.
```
Comment (dannoye) [inline src/_areas/design-system/components/tables.md:14]: A table visualizes and organizes data to help the user derive insights and take an informed action.

File: src/_areas/design-system/components/tables.md [hunk 2]
```diff
@@ -11,75 +11,243 @@ statusOverride: null
...
 
-* Table Header height - 60px
-* Column header row height - 40px
-* Data row height - 48px
-* Table padding - 20px left and right
-* Cell padding - 15px
+##### No results
 
-![Base table spacing and padding](/images/base-table-spacing-padding.png "Base table spacing and padding")
+If there are no results returned from a search, inform the user with a message.
```
Comment (dannoye) [inline src/_areas/design-system/components/tables.md:94]: You could probably reference a link to the Empty states documentation here

File: src/_areas/design-system/components/tables.md [hunk 3]
```diff
@@ -11,75 +11,243 @@ statusOverride: null
...
+#### Actions
 
-![Base table column dividers](/images/column-dividers.png "Base table column dividers")
+##### Table level actions
 
-#### Row banding
+Actions that directly impact the table, such as "Add", can be located in the table header.
 
-In most cases avoid banding.  They can distort the meaning of the data by highlighting every other row to the detriment of neighboring rows and can cause user distraction.  If the rows become hard to follow, particularly when the information is complex, there is considerable space between columns and/or the user is required to scroll horizontally, adding banding will help. 
+* Actions in the header are vertically centered and located on the right side of the header area following the 20px padding of the table
```
Comment (dannoye) [inline src/_areas/design-system/components/tables.md:103]: I'm not sure I understand, "actions in header are vertically centered". I think I'd remove this because the dev specs will show the button being centered vertically within the table header

File: src/_areas/design-system/components/tables.md [hunk 4]
```diff
@@ -11,75 +11,243 @@ statusOverride: null
...
+* Each row, and all column data associated with that row, will appear on a single card. 
+* Row-level actions will be place below a separator line on the bottom-right of each card
+* Each card should be full width to the edge of the screen and have 10px above and below
+
+![Example of a responsive table.](/images/table-mobile.svg "Example of a responsive table.")
+
+<hr>
+
+## Best practices
+
```
Comment (dannoye) [inline src/_areas/design-system/components/tables.md:244]: Idea: Show Do not: Use primary or secondary buttons in table row level action. Do: use link button....etc.

File: src/_areas/design-system/components/tables.md [hunk 5]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+* Depending on the time necessary to load additional rows, a loading experience may be necessary.
+
+![Example of a table with infinite scrolling.](/images/table-infinitescroll.svg "Example of a table with infinite scrolling.")
+
+#### Standard pagination
+
+When infinite scrolling is not an option, a pager may be used to allow the user to navigate through the table.
+
+* Pagers are located 10px directly below the table
+* The pager should indicate the page the user is currently viewing and provide them with 5 other pages to click on including the last page. This will also give the user an indication of how many pages there are total.
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:159]: This is confusing to me. We might want to think of how to reword this a bit.

Comment (itsmemin) [reply src/_areas/design-system/components/tables.md:159]: On second thought, I think this can just be pointed to the pager component. 
https://booster.zywave.dev/design-system/components/pagers/?tab=usage

File: src/_areas/design-system/components/tables.md [hunk 6]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+* The input to the right of the pages allows the user to easily enter the page number they wish to navigate too.
+
+![Example of a table with standard and minimal pagination. ](/images/table-pagination.svg "Example of a table with standard and minimal pagination. ")
+
+#### B﻿est practices
+
+* W﻿hen the user is trying to find something specific,  such as a specific account, a pager is ideal.
+* W﻿hen the user is browsing, such as a content search, infinite scroll may be ideal.
+* F﻿or mobile, infinite scrolling is ideal.
+* If the user navigates to a page from within the table, then uses breadcrumbs to navigate back, we should try to remember the location.
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:182]: This section feels a bit redundant. Can we put these details in the sections above?

File: src/_areas/design-system/components/tables.md [hunk 7]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+
+#### Wrapping data
+
+The benefits of wrapping data is that all of the information is visible to the user and it works well with very detailed information such as plan names
+
+The negative of wrapping data is that wrapping on more than two lines can add a lot of height to the rows within a table, which can actually harm readability of some data like numbers
+
+#### Truncating data
+
+The benefit of truncating data is that it can keep the table compact in terms of height
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:198]:
```suggestion
The benefit of truncation is that the table height remains compact and all row heights will remain equal. 
```

File: src/_areas/design-system/components/tables.md [hunk 8]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+* F﻿or mobile, infinite scrolling is ideal.
+* If the user navigates to a page from within the table, then uses breadcrumbs to navigate back, we should try to remember the location.
+
+### Wrapping and truncation
+
+Based on the space available and the amount of data contained within the table, wrapping or truncation may be necessary to display all of the available columns. The data contained within the table should drive the decision on whether to truncate or wrap. 
+
+![Example of a table with wrapping and truncation.](/images/table-wrappingandtruncation.svg "Example of a table with wrapping and truncation.")
+
+#### Wrapping data
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:190]: Should these be bullet points? If yes, they don't need a . at the end. 
If no, then include a .

File: src/_areas/design-system/components/tables.md [hunk 9]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+
+The benefits of wrapping data is that all of the information is visible to the user and it works well with very detailed information such as plan names
+
+The negative of wrapping data is that wrapping on more than two lines can add a lot of height to the rows within a table, which can actually harm readability of some data like numbers
+
+#### Truncating data
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:196]: Same question here about bullet points.

File: src/_areas/design-system/components/tables.md [hunk 10]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+* W﻿hen the user is trying to find something specific,  such as a specific account, a pager is ideal.
+* W﻿hen the user is browsing, such as a content search, infinite scroll may be ideal.
+* F﻿or mobile, infinite scrolling is ideal.
+* If the user navigates to a page from within the table, then uses breadcrumbs to navigate back, we should try to remember the location.
+
+### Wrapping and truncation
+
+Based on the space available and the amount of data contained within the table, wrapping or truncation may be necessary to display all of the available columns. The data contained within the table should drive the decision on whether to truncate or wrap. 
+
+![Example of a table with wrapping and truncation.](/images/table-wrappingandtruncation.svg "Example of a table with wrapping and truncation.")
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:188]: Can we separate these images and include them in their respective sections below?

File: src/_areas/design-system/components/tables.md [hunk 11]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+
+The negative of wrapping data is that important information may be hidden
+
+#### Things to consider
+
+* When truncating data a tooltip may be used on hover to show the full length
+* Wrapping to a certain number of lines before truncating may be appropriate
+* Move to a full-width table when many columns start wrapping, but consider how this will behave at different browser sizes
+
+### Sorting
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:208]: Move this into the Types section

File: src/_areas/design-system/components/tables.md [hunk 12]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+
+![Example of a table with sorting. ](/images/table-sorting.svg "Example of a table with sorting. ")
+
+Allow users to organize rows by a specific category and follow these best practices: 
+
+* Any column can be ordered alphabetically or numerically by clicking the sort icon next to the column label.
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:214]:
```suggestion
* Any column can be ordered alphabetically, by date, or numerically by clicking the sort icon next to the column label.
```

File: src/_areas/design-system/components/tables.md [hunk 13]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+* Whenever a column is selected for sort, sort by that column only.
+* To help users sort information, a column can be displayed sorting by default. To indicate which column is sorted by default, place a ZUI-chevron up or ZUI-chevron down icon to the right of the column header’s name.
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:216]:
```suggestion
* When helpful, a column can be displayed sorting by default
```

File: src/_areas/design-system/components/tables.md [hunk 14]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+* Move to a full-width table when many columns start wrapping, but consider how this will behave at different browser sizes
+
+### Sorting
+
+![Example of a table with sorting. ](/images/table-sorting.svg "Example of a table with sorting. ")
+
+Allow users to organize rows by a specific category and follow these best practices: 
+
+* Any column can be ordered alphabetically or numerically by clicking the sort icon next to the column label.
+* Whenever a column is selected for sort, sort by that column only.
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:215]:
```suggestion
* Whenever a column is selected for sort, sort by that column only
* To indicate columns than have been sorted, the ZUI-chevron-up or ZUI-chevron-down icon will be highlighted in blue 
* *Sorting by the column in ascending order is indicated with a ZUI-chevron up*
* *Sorting by the column in descending order is indicated with a ZUI-chevron down*
```

File: src/_areas/design-system/components/tables.md [hunk 15]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+* To help users sort information, a column can be displayed sorting by default. To indicate which column is sorted by default, place a ZUI-chevron up or ZUI-chevron down icon to the right of the column header’s name. 
+* *Sorting by the column in ascending order is indicated with a ZUI-chevron up.*
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:217]:
```suggestion
```

File: src/_areas/design-system/components/tables.md [hunk 16]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+* *Sorting by the column in descending order is indicated with a ZUI-chevron down.*
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:218]:
```suggestion

```

File: src/_areas/design-system/components/tables.md [hunk 17]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+* The clickable area to trigger sort, is the whole column header.
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:219]:
```suggestion
* The entire column header is clickable to trigger sorting
```

File: src/_areas/design-system/components/tables.md [hunk 18]
```diff
@@ -11,75 +11,246 @@ statusOverride: null
...
+Allow users to organize rows by a specific category and follow these best practices: 
+
+* Any column can be ordered alphabetically or numerically by clicking the sort icon next to the column label.
+* Whenever a column is selected for sort, sort by that column only.
+* To help users sort information, a column can be displayed sorting by default. To indicate which column is sorted by default, place a ZUI-chevron up or ZUI-chevron down icon to the right of the column header’s name. 
+* *Sorting by the column in ascending order is indicated with a ZUI-chevron up.*
+* *Sorting by the column in descending order is indicated with a ZUI-chevron down.*
+* The clickable area to trigger sort, is the whole column header. 
+
+### Filtering
```
Comment (itsmemin) [inline src/_areas/design-system/components/tables.md:221]: We have multiple different ways to handle filtering. We should cover all of those. 

- Dropdown button (Simple filtering with just a few options)
- Button to open a dialog
- Header filters (again, this is better with only a few options)
- Filter panel

File: src/_areas/design-system/components/tables.md [hunk 19]
```diff
@@ -11,75 +11,335 @@ statusOverride: null
 ---
 ## Usage
```
Comment (dannoye) [inline src/_areas/design-system/components/tables.md:12]: Add a subtitle under the title to match our other components and patterns. 

Ex: Dropdown select Subtitle: Type of form control where a user is presented with a list of options to choose one items from. Usage: A form control used when there are more than 2 options and may require a user to select 1 option.

File: src/_areas/design-system/components/tables.md [hunk 20]
```diff
@@ -11,75 +11,335 @@ statusOverride: null
...
 
-* Generally, you should align text left and numbers right if the numbers are going to be added up or compared to each other
-* Right align numbers when the are monetary values, this allows you to easily compare values and the decimal point lines up
-* Leave text ragged-right, do not justify
-* Year, date, and time should typically be right aligned
+![Anatomy of a table](/images/table-anatomy.svg "Anatomy of a table")
+
+Tables always use cards as their container(s).
+
+* **Table header** (optional): The container that can include search, filters, actions, number of results
```
[the remark this answers is not in the document]
Comment (jsayles12) [reply src/_areas/design-system/components/tables.md:26]: Good catch. It definitely makes sense. I just made the update.
